ShowerShow/Control/UserPreferencesController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using ShowerShow.Controllers;
+using ShowerShow.DTO;
+using ShowerShow.Models;
+using ShowerShow.Repository.Interface;
+using ShowerShow.Service;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShowerShow.Control
+{
+    public class UserPreferencesController
+    {
+        private readonly ILogger<UserPreferencesController> _logger;
+        private IUserPrefencesService _userPrefencesService;
+        private IUserService _userService;
+        public UserPreferencesController(ILogger<UserPreferencesController> logger, IUserPrefencesService userPrefencesService, IUserService userService)
+        {
+            _logger = logger;
+            this._userPrefencesService = userPrefencesService;
+            this._userService = userService;
+
+        }
+
+        [Function("CreateUserPreferences")]
+        [OpenApiOperation(operationId: "CreatePreferences", tags: new[] { "Preferences" })]
```

Comment:
Add summary and description here. Also add summary and description to all your endpoints for more clearance.